### PR TITLE
Documentation for Installing Mathsar using Docker Image

### LINKED ISSUE #2783: Add Documentation for setting up Mathesar with the Mathesar Service Docker Image
## Problem
Users want to use Mathesar with Kubernetes, Nomad, etc. Our docker-compose script isn't helpful in such cases. We need to provide the users an option to run a barebone version of Mathesar 

## Proposed solution
Add Documentation for setting up Mathesar with the Mathesar Service Docker Image.

### PR DESCRIPTION
**Pull request by silentninja**
Fixes #2783 

Adds instructions for running the base Mathesar Docker Image.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by silentninja** [comment]
The PR is ready to be reviewed but I marked it as a draft because it depends on #2826 (I made some branching mistakes while working in parallel and ended up committing to the wrong branch)

File: docs/docs/install/docker/index.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+1. Run the Mathesar Docker Image
+    ```bash
+    docker run \
+      --detach
+      -e DJANGO_DATABASE_URL='<replace with a postgres connection string>' \
+      -e DJANGO_DATABASE_KEY='default' \
+      -e MATHESAR_DATABASES='(<unique_db_key>|<replace with a postgres connection array>)' \
+      -e SECRET_KEY='<replace with a 50 character string>' \
+      -e ALLOWED_HOSTS='.localhost, 127.0.0.1, [::1]' \
+      -e DJANGO_SETTINGS_MODULE='config.settings.production' \
```
Comment (pavish) [inline docs/docs/install/docker/index.md:30]: Are `DJANGO_DATABASE_KEY` and `DJANGO_SETTINGS_MODULE ` essential here?

I think the more minimal the command is, the better it is. Anyone who wants to customize, can refer the configuration guidelines.

File: docs/docs/install/docker/index.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    
+        docker run \
+          -d \
+          # We haven't yet deleted the old Mathesar container so you need to start this new one
+          # with a different name to prevent an error like:
+          # `response from daemon: Conflict. The container name "/mathesar_service" is already in use` 
+          --name mathesar_service_version_REPLACE_WITH_NEW_VERSION \
+          # YOUR STANDARD ARGS HERE
+          mathesar/mathesar-prod:REPLACE_WITH_LATEST_VERSION
+    ```
```
Comment (pavish) [inline docs/docs/install/docker/index.md:95]: Wouldn't stopping the service, pulling the latest image, and starting it again, ensure that the latest image is used?

Why would the user have to manually create a new container with a different name?

File: docs/docs/install/docker/index.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+2. Verify if the Mathesar Server is running successfully:
+    ```bash
+    docker logs -f mathesar_service
+    ```
+
+3. Run Mathesar Installation script
+    ```bash
+    # Install Mathesar types and casting function on the databases specified using the `MATHESAR_DATABASES` env variable in the previous step, run database migrations on the Django meta database, and store the static files in the `static` volumes.
+    docker exec mathesar_service python install.py --skip-confirm
```
Comment (pavish) [inline docs/docs/install/docker/index.md:62]: I think we need more information for this step: all the changes that happen to the database that the user connects to.

**Review by pavish — CHANGES_REQUESTED** [review]

File: docs/docs/install/docker/index.md (hunk 2 above)
Comment (silentninja) [reply docs/docs/install/docker/index.md:95]: No docker won't do it automatically. If the same container name is used for a different image, it throws up the `Conflict. The container name "/mathesar_service" is already in use` which I have mentioned in the comment

**Review by mathemancer — DISMISSED** [review]
I think it's great overall, and this would be useful for some users. We need to come up with a name and structure for this (in combination with the other installation method) that makes the actual distinction between them clear. See my specific comment.

**Comment by seancolsen** [comment]
@silentninja I'd like to hold off on reviewing this until you've made the changes to address other reviews.

**Review by seancolsen — COMMENTED** [review]
I'm mostly reviewing this for language, formatting, and readability. To that end, I pushed ee07a4bd96154a18864be7d6d5445b3b872445da with some minor cleanup.

I'm not reviewing for accuracy or content though, because I don't think I understand this stuff well enough.

However I will say:

- On the "Install Mathesar" page, I think we need to give the user more guidance in choosing an installation method now that we have two methods available. For someone who only vaguely understands what Docker is, how does that person choose between "Docker Compose" and "Docker"? I think that a potentially confusing choice like this so early in the installation process could present a real barrier to some people seeking to install Mathesar. We should explain what the difference is between the installation methods and give some examples for why people might choose particular methods. I would suggest some specific text here, but I don't actually understand the difference between these two methods well enough myself.

**Review by pavish — APPROVED** [review]
I'm going to merge this in because it seems like @mathemancer and @seancolsen have no issues with the content. I think the content looks good too.

All of us seem to find the structure troublesome. Aggregated here:
1. Naming of the Docker installation option
2. Distinction between the docker options
3. Presentation of the installation options on the initial Installation page

I think we can tackle these separately. I've created https://github.com/centerofci/mathesar/issues/2864 for this.

**Pull request merged**